docs/sources/flow/reference/components/otelcol.auth.oauth2.md

```diff
@@ -0,0 +1,121 @@
+---
+title: otelcol.auth.oauth2
+---
+
+# otelcol.auth.oauth2
+
+`otelcol.auth.oauth2` exposes a `handler` that can be used by other `otelcol`
+components to authenticate requests using oauth2 authentication.
+
+The authorization tokens can be used by HTTP and gRPC based OpenTelemetry exporters. 
+This component can fetch and refresh expired tokens automatically. For further details about 
+OAuth2 Client Credentials flow (2-legged workflow) see [here](https://datatracker.ietf.org/doc/html/rfc6749#section-4.4).
+
+> **NOTE**: `otelcol.auth.oauth2` is a wrapper over the upstream OpenTelemetry
+> Collector `oauth2client` extension. Bug reports or feature requests will be
+> redirected to the upstream repository, if necessary.
+
+Multiple `otelcol.auth.oauth2` components can be specified by giving them
+different labels.
+
+## Usage
+
+```river
+otelcol.auth.oauth2 "LABEL" {
+    client_id     = "CLIENT_ID"
+    client_secret = "CLIENT_SECRET"
+    token_url     = "TOKEN_URL"
+}
+```
+
+## Arguments
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`client_id` | `string` | The client identifier issued to the client. | | yes
+`client_secret` | `string` | The secret string associated with the client identifier. | | yes
+`token_url` | `string` | The server endpoint URL from which to get tokens. | | yes
+`endpoint_params` | `map([]string)` | Additional parameters that are sent to the token endpoint. | `{}` | no
+`scopes` | `[]string` | Requested permissions associated for the client. | `[]` | no
+`timeout` | `duration` | The timeout on the client connecting to `token_url` for both initial tokens and refreshing tokens. `"0s"` implies no timeout. | `"0s"` | no
```

Comment:
Can you make this shorter and move the longer description outside of the table? The description in the table should be a short description, and attribute behavior should be documented outside of it. (This is something I'll document)